src/Umbraco.Infrastructure/Packaging/PackageDataInstallation.cs

```diff
@@ -1243,18 +1250,47 @@ public IReadOnlyList<ILanguage> ImportLanguages(IEnumerable<XElement> languageEl
         /// <param name="macroElements">Xml to import</param>
         /// <param name="userId">Optional id of the User performing the operation</param>
         /// <returns></returns>
-        public IReadOnlyList<IMacro> ImportMacros(IEnumerable<XElement> macroElements, int userId)
+        public IReadOnlyList<IMacro> ImportMacros(
+            IEnumerable<XElement> macroElements,
+            IEnumerable<XElement> macroPartialViewsElements,
+            int userId)
         {
             var macros = macroElements.Select(ParseMacroElement).ToList();
 
-            foreach (var macro in macros)
+            foreach (IMacro macro in macros)
             {
                 _macroService.Save(macro, userId);
             }
 
+            ImportMacroPartialViews(macroPartialViewsElements);
+
             return macros;
         }
 
+        private void ImportMacroPartialViews(IEnumerable<XElement> viewElements)
+        {
+            foreach (XElement element in viewElements)
+            {
+                var path = element.AttributeValue<string>("path");
+                if (path == null)
+                {
+                    throw new InvalidOperationException("No path attribute found");
+                }
```

Comment:
An exception is needed here because the XML contract will have failed. If the attribute is not there then this package wasn't created correctly. However, the other exception for having no content can be changed to be the same as partial views (i.e. we'll accept having an empty view).